src/Compilers/CSharp/Test/Emit3/Semantics/PatternMatchingTests4.cs

```diff
@@ -4829,5 +4842,4429 @@ class D : A { }
+                // _ = i is not (42 or 43) or (43 or 0); // warn
+                Diagnostic(ErrorCode.WRN_RedundantPattern, "0").WithLocation(6, 35),
+                // (7,35): warning CS9275: The pattern is redundant.
+                // _ = i is not (42 or 43) or 43 or (0 or 1); // warn
+                Diagnostic(ErrorCode.WRN_RedundantPattern, "0").WithLocation(7, 35),
+                // (7,40): warning CS9275: The pattern is redundant.
+                // _ = i is not (42 or 43) or 43 or (0 or 1); // warn
+                Diagnostic(ErrorCode.WRN_RedundantPattern, "1").WithLocation(7, 40),
+                // (8,36): hidden CS9274: The pattern is redundant.
+                // _ = i is (not (42 or 43) or 43) or 0;
+                Diagnostic(ErrorCode.HDN_RedundantPattern, "0").WithLocation(8, 36),
+                // (9,35): warning CS9275: The pattern is redundant.
+                // _ = i is not (42 or 43) or (43 or 0); // warn
+                Diagnostic(ErrorCode.WRN_RedundantPattern, "0").WithLocation(9, 35),
+                // (10,29): warning CS9275: The pattern is redundant.
+                // _ = i is not (42 or 43) or (0 or 43); // warn
+                Diagnostic(ErrorCode.WRN_RedundantPattern, "0").WithLocation(10, 29),
+                // (11,45): warning CS9275: The pattern is redundant.
+                // _ = i is string or not (42 or 43) or (43 or 0); // warn
+                Diagnostic(ErrorCode.WRN_RedundantPattern, "0").WithLocation(11, 45),
+                // (12,44): warning CS9275: The pattern is redundant.
+                // _ = i is string or not (42 or 43) or 43 or 0; // warn
+                Diagnostic(ErrorCode.WRN_RedundantPattern, "0").WithLocation(12, 44),
+                // (13,38): warning CS9275: The pattern is redundant.
+                // _ = i is string or not (42 or 43) or 0; // warn
+                Diagnostic(ErrorCode.WRN_RedundantPattern, "0").WithLocation(13, 38),
+                // (14,32): hidden CS9274: The pattern is redundant.
+                // _ = i is (string or not 42) or 0;
+                Diagnostic(ErrorCode.HDN_RedundantPattern, "0").WithLocation(14, 32),
+                // (15,17): hidden CS9274: The pattern is redundant.
+                // _ = i is 42 or (42 or not 0);
+                Diagnostic(ErrorCode.HDN_RedundantPattern, "42").WithLocation(15, 17),
+                // (16,22): hidden CS9274: The pattern is redundant.
+                // _ = i is (not 42) or 43;
+                Diagnostic(ErrorCode.HDN_RedundantPattern, "43").WithLocation(16, 22));
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/75506")]
+        public void RedundantPattern_SyntaxHeuristic_Hidden()
+        {
+            var source = """
+// We're okay reporting the following as just hidden diagnostics
+C s = null;
+_ = s is { Prop1: not 42 } or { Prop1: 43 }; // 1
+_ = s is (not 42, _) or (not 43, _); // 2
```

Comment:
I find it confusing that this line does `not 43` while the previous line does just `43`. Maybe this avoids "multiplying out" the combinations?

Pausing here till after lunch.

Reply:
Yes, it's introducing some variation without a separate test